sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchema.scala

```diff
@@ -40,7 +41,19 @@ private[csv] object CSVInferSchema {
       csv: Dataset[String],
       caseSensitive: Boolean,
       options: CSVOptions): StructType = {
-    val firstLine: String = CSVUtils.filterCommentAndEmpty(csv, options).first()
+    val lines = CSVUtils.filterCommentAndEmpty(csv, options)
```

Comment:
Hi @wojtek-szymanski I think we should not rely on exception handling. I can think of `take(1).headOption` but we could use shorten one if you know any other good way. What do you think about this?